docs/markdown/DevGuide_BuildProcessor.md

```diff
@@ -0,0 +1,115 @@
+    : Processor()
+    , inport("imageInport")
+    , myProp("myProp", "My Property", 0.5f, 0.0f, 1.0f) {
+        addPort(inport);
+        addProperty(myProp);
+}
+```
+Note that the constructor of the `FloatProperty` takes an identifier, display name, default, min and max value as parameters. You can then use `myProp` in your code as you would use a normal `float`. The `ImageInport` only requires an identifier.
+
+
+## Inports and Outports
+Similar to properties, you can add ports as private members of your class and initialize them in the constructor. Ports also need to be added to the processor using `addPort()`. Note that the convention is to put optional inports to the right of the processor (added last). If your processor performs some kind of rendering, it should usually also include an optional `ImageInport`, where previously rendered parts of an image can be passed through, so that the processor's outport produces a composited image of the current and previous renderings.
+
+## process()
+The `process()` method is where the functionality of your processor is defined. Every time the processor is invalidated (either by changing inputs, events or property changes), the `process()` method is executed.
+Inside `process()` you can access your inports' data by using `yourinport.getData()` and similarly you can write data to the outports using `youroutport.setData(...)`. What happens with your data in between is fully up to you.
+You can also use all your defined properties here. This let's you access all your algorithm's parameters directly from the GUI with automatic updates upon change.
+
+### Frequently used operations
+
+#### Setting up an `ImageOutport`:
+As mentioned above, most processors that output an image should also have an optional image inport for compositing of different renders. Usually by default you want to render on top of this incoming image, which basically holds an OpenGL framebuffer. That means you usually want to copy the inport data to the outport and activate it as target framebuffer for your rendering. `utilgl::activateTargetAndClearOrCopySource(outport, image_inport)` does exactly this for you. All your rendering will be automatically composited according to the framebuffer depth layer. Note that it is good practice to deactivate the target at the end of your `process()` by using `utilgl::deactivateCurrentTarget()`. You can include the function from `modules/opengl/texture/textureutils.h`.
+
+#### Shader and Texture utilities:
+The `utilgl` namespace includes more useful functions for OpenGL, like controlling various OpenGL states (see `modules/opengl/openglutils.h`), utilities for working with `Shader`s (see `modules/opengl/shader/shaderutils/h`), drawing full screen quads and more.
+
+The `Shader` class can be used to activate shaders and bind uniforms etc. Similarly you can use the `TextureUnit` class as a wrapper for texture units. An image inport can be bound to a texture unit using
```

Comment:
Here we could also point the reader to a processor. Perhaps we also need to mention TextureUnitContainer